functions-continued/variable-scope.md

```diff
@@ -0,0 +1,396 @@
+print("Instead of trying to access breakfast_message, let's return breakfast_message, store it in resulting_message, and access resulting_message instead:")
+print(f"The value of resulting_message is {resulting_message}")
+```
+
+**This will be one of the main patterns that we use in programming.**
+
+### Recognizing Variable Scope
+
+Beyond functions, other kinds of **blocks of code** in Python also create variable scope. We will learn about those in detail in the future.
+
+For now, here is **one tip**: there is **a pattern** between **local variable scope** and **its level of indentation** in code.
+
+## There Are More Kinds of Variables
+
+**Global variables** are a kind of variable with global variable scope. Global variables can be accessed inside **or** outside of any functions.
+
+Typically, global variables are declared **outside of any functions.**
+
+Properties of global variables:
+
+- named with the `lower_snake_case` naming convention we've been using so far already.
+- variables with the largest scope.
+  - Can be accessed and modified outside any function
+  - Can be accessed and modified inside any function
+- a kind of variable that we will use commonly at the beginning of our coding journey, and will use often while debugging or during in-progress code.
+  - Using global variables right now is inevitable
+  - Over time, we'll learn more tools that will help us avoid using global variables, or increase our code quality even when we do use global variables.
+
+```python
+def display_breakfast():
+    print("Let's try to access the value of breakfast_message INSIDE of the display_breakfast function:")
+    print(f"{breakfast_message}")
+
+
+breakfast_message = "My breakfast today is an apple"
+print("Let's try to access the value of breakfast_message OUTSIDE of the display_breakfast function:")
+print(f"The value of breakfast message is {breakfast_message}")
+
+display_breakfast()
+```
+
+In this example, `breakfast_message` is a global variable, because it was declared **outside** of any function in our code. We are able to access it outside of the `display_breakfast` function, AND inside the `display_breakfast` function. We did **not** get a `NameError` when running this code, because `breakfast_message` is in scope.
+
+Because global variable scope is so large and so powerful, global variable scope gives programmers more opportunities to create bugs. Over time, observe what tools and strategies are introduced to help lower this risk!
+
+### !callout-info
+
+## Variables Best Practices
+
+Technically, Python has one specific rule about global variables: we can **read** global variables anywhere. However, **declaring** global variables inside of a function is a little trickier. We discourage declaring global variables inside of functions! Programmers discourage it so much, that this curriculum won't even teach you how to do it! Feel free to look this up. A search engine term to use could be "Python use global variables".
```

Comment:
Manually changing your suggestion to remove the final "But let's try to avoid this in our own code!" -- Within the context of this section, I think it's snappier to not repeat that and leave it as a "You're free to check if you still want to by the end of this paragraph lol"


I'm leaving the phrase "has one specific rule" so the awkward phrasing gives emphasis as a fact to remember. lol /shrug